### DISCUSSION TITLE
Is it possible for a custom Element Type to eager load Records instead of Elements?

### DISCUSSION THREAD

**Discussion by joshuapease** [craftcms/cms#12171] [Q&A] [answered]
I'm working on a project that has a custom Element Type, that has a `hasMany` relationship to some data stored as Active Records. 

It's for a custom quoting system. The Quote is a Custom Element and each quote can have many LineItems (which are Active Records). 

I've read the docs on [eager loading Element Types](https://craftcms.com/docs/3.x/extend/element-types.html#eager-loading) and it seems to imply that only other Element types can be eager loaded with `eagerLoadingMap()`.

Is that a correct assessment? Or are there other ways to eager load "non Elements"  that are related to a custom Element Type? 

----

Edit: Actually... while digging around the discussions, I think I might have answered my question. Seems like `populate()` in my Element's query class might be the place to attach this behavior. 

Figured it might still be worthwhile to post in case other folks have a similar question or there's a better way to approach this.  

**Related Discussion:** 
- https://github.com/craftcms/cms/discussions/11262

**Example in Craft Commerce**
https://github.com/craftcms/commerce/blob/develop/src/elements/db/OrderQuery.php#L1150-L1177

**Comment by brandonkelly** [comment, marked as answer]
The built-in element eager-loading support is limited to elements and we don’t have any plans to change that. You’ll have to do something similar to Commerce orders, as you’ve found.
